### PR TITLE
CLI for NVGRE

### PR DESCRIPTION
**Pull request by vadymhlushko-mlnx**

### CODE REVIEW

File: config/plugins/nvgre_tunnel.py [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+    if src_ip is not None:
+        data["src_ip"] = src_ip
+
+    try:
+        add_entry_validated(db.cfgdb, table, key, data)
+    except Exception as err:
+        exit_with_error(f"Error: {err}", fg="red")
+
+
+@NVGRE_TUNNEL.command(name="update")
```
Comment (vadymhlushko-mlnx) [inline config/plugins/nvgre_tunnel.py]: remove update commands and tests

File: config/plugins/nvgre_tunnel.py [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+    required=True,
+)
+@click.argument(
+    "vlan-name",
+    nargs=1,
+    required=True,
+)
+@click.option(
+    "--vlan-id",
+    help="",
```
Comment (vadymhlushko-mlnx) [inline config/plugins/nvgre_tunnel.py]: add mandatory flag to all parameters

File: tests/nvgre_tunnel_test.py [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+        
+        # delete
+        result = runner.invoke(
+            config.config.commands["nvgre-tunnel"].commands["delete"],
+            ["tunnel_1"], obj=db
+        )
+
+        logger.debug("\n" + result.output)
+        logger.debug(result.exit_code)
+        assert result.exit_code == SUCCESS
```
Comment (vadymhlushko-mlnx) [inline tests/nvgre_tunnel_test.py:76]: add verify for show output

File: tests/nvgre_tunnel_test.py [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
```
Comment (vadymhlushko-mlnx) [inline tests/nvgre_tunnel_test.py:117]: add verify for show

File: tests/nvgre_tunnel_test.py [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+
+        result = runner.invoke(
+            config.config.commands["nvgre-tunnel"].commands["delete"],
+            ["tunnel_2"], obj=db
+        )
+
+        logger.debug("\n" + result.output)
+        logger.debug(result.exit_code)
+        assert result.exit_code == SUCCESS
```
Comment (vadymhlushko-mlnx) [inline tests/nvgre_tunnel_test.py:126]: add verify for show

File: tests/nvgre_tunnel_test.py [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+
+
+    def test_nvgre_tunnel_update(self):
```
Comment (vadymhlushko-mlnx) [inline tests/nvgre_tunnel_test.py]: remove tets case

File: tests/nvgre_tunnel_test.py [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+
+        # add
+        result = runner.invoke(
+            config.config.commands["nvgre-tunnel"].commands["add"],
+            ["tunnel_1", "--src-ip", INVALID_VALUE], obj=db
+        )
+
+        logger.debug("\n" + result.output)
+        logger.debug(result.exit_code)
+        assert result.exit_code == ERROR
```
Comment (vadymhlushko-mlnx) [inline tests/nvgre_tunnel_test.py:144]: add verify for show

File: tests/nvgre_tunnel_test.py [hunk 8]
```diff
@@ -0,0 +1,274 @@
...
+            config.config.commands["nvgre-tunnel-map"].commands["add"],
+            ["tunnel_1", "Vlan2000", "--vlan-id", "2000", "--vsid", "6000"], obj=db
+        )
+
+        logger.debug("\n" + result.output)
+        logger.debug(result.exit_code)
+        assert result.exit_code == SUCCESS
+
+        # verify
+        result = runner.invoke(show.cli.commands["nvgre-tunnel-map"], [], obj=db)
```
Comment (vadymhlushko-mlnx) [inline tests/nvgre_tunnel_test.py]: create function for verify

File: tests/nvgre_tunnel_test.py [hunk 9]
```diff
@@ -0,0 +1,274 @@
...
+
+        # delete
+        result = runner.invoke(
+            config.config.commands["nvgre-tunnel-map"].commands["delete"],
+            ["tunnel_1", "Vlan1000"], obj=db
+        )
+
+        logger.debug("\n" + result.output)
+        logger.debug(result.exit_code)
+        assert result.exit_code == SUCCESS
```
Comment (vadymhlushko-mlnx) [inline tests/nvgre_tunnel_test.py:188]: add verify for show cmd

File: tests/nvgre_tunnel_test.py [hunk 10]
```diff
@@ -0,0 +1,274 @@
...
+
+    def test_nvgre_tunnel_map_add_invalid_vsid(self):
+        dbconnector.dedicated_dbs['CONFIG_DB'] = os.path.join(mock_db_path, 'nvgre_tunnel')
+        db = Db()
+        runner = CliRunner()
+
+        # add
+        result = runner.invoke(
+            config.config.commands["nvgre-tunnel-map"].commands["add"],
+            ["tunnel_1", "Vlan1000", "--vlan-id", "1000", "--vsid", "1"], obj=db
```
Comment (vadymhlushko-mlnx) [inline tests/nvgre_tunnel_test.py]: seems like value `1` valid add test case to check vsid range

File: tests/nvgre_tunnel_test.py [hunk 11]
```diff
@@ -0,0 +1,274 @@
...
+        )
+
+        logger.debug("\n" + result.output)
+        logger.debug(result.exit_code)
+        assert result.exit_code == ERROR
+
+
+    def test_nvgre_tunnel_map_update(self):
```
Comment (vadymhlushko-mlnx) [inline tests/nvgre_tunnel_test.py]: remove test case

**Review by ayurkiv-nvda — APPROVED** [review]

**Review by volodymyrsamotiy — APPROVED** [review]

**Pull request closed without merging**